zipline/finance/blotter.py

```diff
@@ -329,20 +329,23 @@ def get_transactions(self, bar_data):
                     if not order.open:
                         closed_orders.append(order)
 
+        self.prune_orders(closed_orders)
+
+        # FIXME this API doesn't feel right (returning two things here)
+        return transactions, None
+
+    def prune_orders(self, closed_orders):
         # remove all closed orders from our open_orders dict
         for order in closed_orders:
             sid = order.sid
             try:
                 sid_orders = self.open_orders[sid]
                 sid_orders.remove(order)
-            except KeyError:
+            except ValueError:
```

Comment:
> Either way, we're trying to remove something that is already removed, so we shouldn't blow up on it.

That's fair.  Just wondering if there's some bad state tracking elsewhere, in that we're trying to remove it at all, and whether that should be fixed instead, as opposed to fixing all those downstream of it.